src/pages/TodoList/index.tsx

```diff
@@ -0,0 +1,169 @@
+import { useEffect, useState } from 'react';
+import { DragDropContext, Droppable, Draggable } from 'react-beautiful-dnd';
+import classNames from 'classnames';
+import { useAppDispatch, useAppSelector } from '@/store';
+import {
+  Todo, TodoType,
+} from '@/types';
+import { clearCompleted, setSelectedTodoType, updateTodoList } from '@/features/todo/todoSlice';
+import styles from './todolist.module.scss';
+import { TodoItem } from '@/components/molecules';
+import { Button } from '@/components/atoms';
+
+const TodoList = () => {
+  const todoState = useAppSelector((state) => state.todoState);
+  const { todos, selectedTodoType } = todoState;
+
+  const dispatch = useAppDispatch();
+
+  const [leftItemCount, setLeftItemCount] = useState(0);
+
+  const [listedRecords, setListedRecords] = useState<Todo[]>([]);
+
+  const calculateLeftItemCount = () => {
+    if (selectedTodoType === TodoType.ACTIVE) {
+      setLeftItemCount(listedRecords.filter((todo:Todo) => !todo.completed).length);
+    } else if (selectedTodoType === TodoType.COMPLETED) {
+      setLeftItemCount(listedRecords.filter((todo:Todo) => todo.completed).length);
+    } else {
+      setLeftItemCount(listedRecords.length);
+    }
+  };
+
+  /**
+   *
+   */
+  const listAllRecors = () => {
+    dispatch(setSelectedTodoType(TodoType.ALL));
+    setListedRecords([...todos]);
+  };
+
+  /**
+   *
+   */
+  const listActiveRecors = () => {
+    dispatch(setSelectedTodoType(TodoType.ACTIVE));
+    const records = [...todos.filter((item:Todo) => item.completed === false)];
+    setListedRecords(records);
+  };
+
+  /**
+   *
+   */
+  const listCompletedRecors = () => {
+    dispatch(setSelectedTodoType(TodoType.COMPLETED));
+    const records = [...todos.filter((item:Todo) => item.completed === true)];
+    setListedRecords(records);
+  };
+
+  /**
+   *
+   */
+  const clearCompletedHandle = () => {
+    dispatch(clearCompleted());
+    dispatch(setSelectedTodoType(TodoType.ACTIVE));
+  };
+
+  const actions:any = {
+    [`${TodoType.ALL}`]: listAllRecors,
+    [`${TodoType.ACTIVE}`]: listActiveRecors,
+    [`${TodoType.COMPLETED}`]: listCompletedRecors,
+  };
+
+  /**
+   *
+   */
+  useEffect(() => {
+    dispatch(setSelectedTodoType(TodoType.ALL));
+  }, []);
+
+  /**
+   *
+   */
+  useEffect(() => {
+    calculateLeftItemCount();
+  }, [listedRecords]);
+
+  /**
+   *
+   */
+  useEffect(() => {
+    if (selectedTodoType) {
```

Comment:
This `useEffect` is not needed. Seems like you are calling function twice:
1. After click
2. After selectedTodoType changes (it happens right after step 1. )

Reply:
it necessary because i need to load data when component created

Reply:
useEffect should be with empty list of dependencies to be called only on mounting

Reply:
You can do something like 
`const listed = selectedTodoType === TodoType.ALL ? todos : todos.filter((item: Todo) => item.completed === (selectedTodoType === TodoType.COMPLETED));`
In this case you will no need to have this Effect, functions `list..`, variable `actions` and you can count left items just by taking a length of `listed`

Reply:
it is fixed;

yes, after doing as you said, the number of lines of code decreased by half, thanks.